src/box/iproto.cc

```diff
@@ -2138,6 +2138,39 @@ tx_process_rollback(struct cmsg *m)
 	tx_end_msg(msg, &header);
 }
 
+/*
+ * In case the request does not contain a space or identifier but contains a
+ * corresponding name, tries to resolve the name.
+ * A space identifier is mandatory for a CRUD request, so in case the space
+ * identifier is not present in the request and we fail to get an identifier
+ * from the space name (or the space name is missing), we must return with an
+ * error, setting a missing request field diagnostic, just as it was done in
+ * `xrow_decode_dml` before space names in IPROTO requests were introduced.
+ */
+static int
+tx_resolve_space_and_index_name(struct request *dml)
+{
+	if (dml->space_id == 0 && dml->space_name != NULL) {
+		uint32_t space_id = box_space_id_by_name(dml->space_name,
```

Comment:
Why do you not use `space_cache_find` here? It's more lightweight and should be faster than the public API function. In general, in the core code we should prefer internal function over public API functions.

Reply:
`space_cache_find` has an inconvenient interface: it only support NULL-terminated strings. Guess I will have to refactor it to allow arbitrary strings :slightly_frowning_face:

Reply:
Fixed, now I use `space_by_name`.